adapters/frvradn/frvradn.go

```diff
@@ -0,0 +1,151 @@
+package frvradn
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/prebid/openrtb/v19/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"github.com/prebid/prebid-server/util/httputil"
+	"strings"
+)
+
+type adapter struct {
+	uri string
+}
+
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	if config.Endpoint == "" {
+		return nil, errors.New("missing endpoint adapter parameter")
+	}
+
+	bidder := &adapter{
+		uri: config.Endpoint,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var requests []*adapters.RequestData
+	var errs []error
+
+	requestCopy := *request
+	for _, imp := range request.Imp {
+		frvrAdnExt, err := getImpressionExt(&imp)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		if imp.BidFloor > 0 && imp.BidFloorCur != "" && strings.ToUpper(imp.BidFloorCur) != "USD" {
+			convertedValue, err := requestInfo.ConvertCurrency(imp.BidFloor, imp.BidFloorCur, "USD")
+			if err != nil {
+				errs = append(errs, err)
+				continue
+			}
+			imp.BidFloorCur = "USD"
+			imp.BidFloor = convertedValue
+		}
+
+		ext, err := json.Marshal(frvrAdnExt)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		imp.Ext = ext
+
+		requestCopy.Imp = []openrtb2.Imp{imp}
+
+		requestJSON, err := json.Marshal(requestCopy)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		requestData := &adapters.RequestData{
+			Method: "POST",
+			Uri:    a.uri,
+			Body:   requestJSON,
+		}
+		requests = append(requests, requestData)
+	}
+	return requests, errs
+}
+
+func (a *adapter) MakeBids(request *openrtb2.BidRequest, requestData *adapters.RequestData, responseData *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if httputil.IsResponseStatusCodeNoContent(responseData) {
+		return nil, nil
+	}
+
+	if err := httputil.CheckResponseStatusCodeForErrors(responseData); err != nil {
+		return nil, []error{err}
+	}
+
+	if len(responseData.Body) == 0 {
+		return nil, nil
+	}
+
+	var response openrtb2.BidResponse
+	if err := json.Unmarshal(responseData.Body, &response); err != nil {
+		return nil, []error{err}
+	}
+
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(len(request.Imp))
+	bidResponse.Currency = response.Cur
+	var errs []error
+	for _, seatBid := range response.SeatBid {
+		for i, bid := range seatBid.Bid {
+			bidType, err := getBidMediaType(bid)
```

Comment:
you are using `bid` var here that is copy of seatBid.Bid. IMO, it would be slightly more optimised (memory wise) to pass address of seatBid.Bid[i] to getBidMediaType function. So - 
```
...
for i, _ := range seatBid.Bid {
   bidType, err := getMediaType(&seatBid.Bid[i])
...
}
func getBidMediaType(bid *openrtb2.Bid) (openrtb_ext.BidType, error) {
...
}
```